docs/contributing-coding-guidelines.md

```diff
@@ -72,17 +84,72 @@ this situation will be handled one of a few ways:
 
-* If a new dependency or dependency update is required, then a trusted
-  contributor of Wasmtime will be required to perform a vet of the new
-  crate/version. This will be done through a separate PR to Wasmtime so we ask
-  contributors to not run `cargo vet` themselves to get CI passing. Reviewers
-  understand what `cargo vet` failures are on CI and how it doesn't reflect on
-  the quality of the PR itself. Once the reviewer (or another maintainer) merges
-  a PR adding the vet entries necessary for the original contribution it can be
-  rebased to get CI passing.
-
-Note that this process is not in place to prevent new dependencies or prevent
-updates, but rather it ensures that development of Wasmtime is done with a
-trusted set of code that has been reviewed by trusted parties. We welcome
-dependency updates and new functionality, so please don't be too alarmed when
-contributing and seeing a failure of `cargo vet` on CI!
+Dependency additions or updates require action on behalf of project maintainers
+so we ask that you don't run `cargo vet` yourself or update the `supply-chain`
+folder yourself. Instead a maintainer will review your PR and push a commit to
+your PR which performs the vets as necessary. Note that reviewers will be
+verifying that you haven't yourself made changes and that the changes they
+approve to be merged match what they've done. It's still ok to rebase your PR if
+needed, but we ask that if a maintainer has pushed a `cargo vet` entry update
+that you leave it as a separate commit.
+
+### `cargo vet` for Maintainers
+
+Maintainers of Wasmtime are required to explicitly vet and approve all
+dependency updates and modifications to Wasmtime. This means that when reviewing
+a PR you should ensure that contributors are not modifying the `supply-chain`
+directory themselves outside of commits authored by other maintainers. Otherwise
+though to add vet entries this is done through one of a few methods:
+
+* For a PR where maintainers themselves are modifying dependencies the `cargo
+  vet` entries can be included inline with the PR itself by the author. The
+  reviewer knows that the author of the PR is themself a maintainer.
+
+* PRs that "just update dependencies" are ok to have at any time. You can do
+  this in preparation for a future feature or for a future contributor. This
+  more-or-less is the same as the previous categories.
+
+* For contributors who should not add vet entries themselves maintainers should
+  review the PR and then when the PR is ready to be approved the maintainer
+  should push a commit to the contributor's PR directly. Maintainers should
+  check out the PR locally, apply `cargo vet` entries as needed, and then push
+  to the contributor's PR directly.
+
+Note for the last case it's important to ensure that after you push to the PR
+any future updates pushed by the contributor either contain or don't overwrite
+your vet entries. It's recommended to add vet entries just before approval to
```

Comment:
Let's add some explicit note heres too -- maybe something like "... or don't overwrite your vet entries. Also verify that if the PR branch is rebased or force-pushed, the details of your previously pushed vetting remain the same: e.g., versions were not bumped and descriptive reasons remain the same. If pushing a vetting commit to a contributor's PR and also asking for more changes, request that the contributor make the requested fixes in an additional commit rather than force-pushing a rewritten history, so your existing vetting commit remains untouched. These guidelines make it easier to verify no tampering has occurred."